### PR TITLE
PDF format API

### PR DESCRIPTION
**Pull request by bjosttveit**
## Description

Added a new controller with a single endpoint for getting the pdf formatting. Tested manually on the `frontend-test` app. Borrowed a lot from `PdfService`.

## Related Issue(s)
- #153 
- https://github.com/Altinn/app-frontend-react/issues/818

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

**Review by tjololo — CHANGES_REQUESTED** [review]
I think it would be wise to rename the Controller to `PdfController.cs` so other API regarding PDF have a natural home/grouping in this controller. This would not change the API but make it easier to implement new pdf APIs in the future.

We are moving away from `Newtonsoft.Json` and replacing it with `System.Text.Json` we should use `System.Text.Json` in new code.

File: src/Altinn.App.Api/Controllers/PdfFormatController.cs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    [ApiController]
+    [Route("{org}/{app}/instances/{instanceOwnerPartyId:int}/{instanceGuid:guid}/data/{dataGuid:guid}/pdfformat")]
+    public class PdfFormatController : ControllerBase
+    {
+        private readonly IInstance _instanceClient;
+        private readonly IPdfFormatter _pdfFormatter;
+        private readonly IAppResources _resources;
+        private readonly IAppModel _appModel;
+        private readonly IData _dataClient;
+        private readonly ILogger _logger;
```
Comment (tjololo) [inline src/Altinn.App.Api/Controllers/PdfFormatController.cs]: No logging done, remove it until we actually do some logging

**Comment by bjosttveit** [comment]
> I think it would be wise to rename the Controller to `PdfController.cs` so other API regarding PDF have a natural home/grouping in this controller. This would not change the API but make it easier to implement new pdf APIs in the future.

In that case, should I also change the route of the controller to be more general? maybe something like:

`{org}/{app}/instances/{instanceOwnerPartyId:int}/{instanceGuid:guid}/data/{dataGuid:guid}/pdf`

And then change the route of `GetPdfFormat()` to something like:

`{org}/{app}/instances/{instanceOwnerPartyId:int}/{instanceGuid:guid}/data/{dataGuid:guid}/pdf/format`

**Comment by tjololo** [comment]
> > I think it would be wise to rename the Controller to `PdfController.cs` so other API regarding PDF have a natural home/grouping in this controller. This would not change the API but make it easier to implement new pdf APIs in the future.
> 
> In that case, should I also change the route of the controller to be more general? maybe something like:
> 
> `{org}/{app}/instances/{instanceOwnerPartyId:int}/{instanceGuid:guid}/data/{dataGuid:guid}/pdf`
> 
> And then change the route of `GetPdfFormat()` to something like:
> 
> `{org}/{app}/instances/{instanceOwnerPartyId:int}/{instanceGuid:guid}/data/{dataGuid:guid}/pdf/format`

I totally agree, nice catch!

**Comment by bjosttveit** [comment]
Some of the controllers like `ValidateController` do not have a route for the controller, and each individual endpoint specify the entire route, is that maybe more appropriate? Since we do not know whether future APIs will require all of the same parameters?

File: src/Altinn.App.Api/Controllers/PdfController.cs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            var result = new
+            {
+                ExcludedPages = layoutSettings.Pages.ExcludeFromPdf,
+                ExcludedComponents = layoutSettings.Components.ExcludeFromPdf,
+                PageOrder = layoutSettings.Pages.Order,
+            };
+            return Ok(result);
+        }
+    }
+}
```
Comment (tjololo) [inline src/Altinn.App.Api/Controllers/PdfController.cs]: Add trailing linebreak
```suggestion
}

```

**Review by tjololo — APPROVED** [review]
:godmode:

**Pull request merged**